File: Doc-Scanner-X/Doc-Scanner-X/LocalFileViewController.m
//
//  LocalFileViewController.m
//  Doc-Scanner-X
//
//  Created by dynamsoft on 2017/11/27.
//  Copyright © 2017年 com.dynamsoft. All rights reserved.
//

#import "LocalFileViewController.h"

@interface LocalFileViewController ()

@end

@implementation LocalFileViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _tableViewVC = [[LocalFileTableViewController alloc] initWithStyle:UITableViewStylePlain];
    [self pushViewController:_tableViewVC animated:YES];
    MainViewController* parentVC = (MainViewController*)self.parentViewController;
    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"11.0")) {
        UIButton* backButton = [UIButton new];
        [backButton setImage:[UIImage imageNamed:@"icon_back"] forState:UIControlStateNormal];
        [backButton addTarget:parentVC action:@selector(backToHome) forControlEvents:UIControlEventTouchUpInside];
        backButton.imageEdgeInsets = UIEdgeInsetsMake(backButton.imageEdgeInsets.top, -14, backButton.imageEdgeInsets.bottom, backButton.imageEdgeInsets.right);
        _tableViewVC.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithCustomView:backButton];
    } else {
        _tableViewVC.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"icon_back"] style:UIBarButtonItemStylePlain target:parentVC action:@selector(backToHome)];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: DocScannerX/DocScannerX/LocalFileViewController.swift
//
//  LocalFileViewController.swift
//  DCS SuperScanner
//
//  Created by dynamsoft on 2017/11/1.
//  Copyright © 2017年 com.dynamsoft. All rights reserved.
//

import UIKit

class LocalFileViewController: UINavigationController {

    // MARK: - Properties
    
    var tableViewVC: LocalFileTableViewController?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        /// Initialize `tableViewVC` and it's root view controller of UINavigationController.
        tableViewVC = LocalFileTableViewController(style: .plain)
        pushViewController(tableViewVC!, animated: true)
        let parentVC = self.parent as? MainViewController
        if #available(iOS 11, *) {
            let backButton = UIButton(type: .custom)
            backButton.setImage(UIImage(named: "icon_back"), for: .normal)
            backButton.addTarget(parentVC, action: #selector(parentVC?.backToHome), for: .touchUpInside)
            backButton.imageEdgeInsets.left = -14
            tableViewVC!.navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
        } else {
            tableViewVC?.navigationItem.leftBarButtonItem = UIBarButtonItem(image: UIImage(named: "icon_back")?.withRenderingMode(.alwaysOriginal), style: .plain, target: parentVC, action: #selector(parentVC?.backToHome))
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
}
